### PR TITLE
Fix density current tutorial

### PR DESCRIPTION
**Pull request by charleskawczynski**
# Description

This PR fixes the density current tutorial in the docs, and adds it to the list of docs.

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by akshaysridhar** [comment]
Should we delete this duplicate @smarras79 : https://github.com/CliMA/ClimateMachine.jl/blob/master/test/Numerics/DGMethods/compressible_Navier_Stokes/density_current_model.jl

**Comment by charleskawczynski** [comment]
Probably, unless it fits better as a test, in which case we could just delete this from the tutorials.

**Comment by akshaysridhar** [comment]
@charleskawczynski I rebased this following yesterday's docs changes.

File: docs/list_of_tutorials.jl [hunk 1]
```diff
@@ -20,6 +20,7 @@ if generate_tutorials
 
     tutorials = [
         "Atmos" => [
+            "Density current" => "Atmos/densitycurrent.jl",
```
Comment (kmdeck) [inline docs/list_of_tutorials.jl:23]: I think @jiahe23 would then just need to add this to the summary of tutorials page, or, if that PR has already gone through, could you add this to the summary page?

https://github.com/CliMA/ClimateMachine.jl/pull/1575

**Comment by jiahe23** [comment]
I got the following error msg when making the docs locally.

```
[ Info: Populate: populating indices.
ERROR: LoadError: `makedocs` encountered an error. Terminating build
Stacktrace:
 [1] [login2:166665:0:166665] Caught signal 7 (Bus error: nonexistent physical address)
[login2:166665:0:166665] mmap(NULL, 4096, READ|WRITE, PRIVATE|ANON) failed: Cannot allocate memory
[login2:166665:0] Caught signal 11 (Segmentation fault)
==== backtrace ====
 2 0x000000000006ba2c mxm_handle_error()  /var/tmp/OFED_topdir/BUILD/mxm-3.7.3111/src/mxm/util/debug/debug.c:641
 3 0x000000000006bf7c mxm_error_signal_handler()  /var/tmp/OFED_topdir/BUILD/mxm-3.7.3111/src/mxm/util/debug/debug.c:616
 4 0x0000000000036280 killpg()  ??:0
 5 0x000000000006b8a0 mxm_debug_get_line_info()  /var/tmp/OFED_topdir/BUILD/mxm-3.7.3111/src/mxm/util/debug/debug.c:340
 6 0x00000000000472da backtrace_next()  ??:0
 7 0x00000000000483bf ucs_handle_error()  ??:0
 8 0x0000000000048750 ucs_handle_error()  ??:0
 9 0x00000000000488a2 ucs_handle_error()  ??:0
10 0x000000000000f5d0 _L_unlock_13()  funlockfile.c:0
11 0x000000000015940b __memmove_ssse3_back()  :0
12 0x0000000001c830b5 _ZN4llvm15RuntimeDyldImpl11emitSectionERKNS_6object10ObjectFileERKNS1_10SectionRefEb()  ??:0
13 0x0000000001c838d5 _ZN4llvm15RuntimeDyldImpl17findOrEmitSectionERKNS_6object10ObjectFileERKNS1_10SectionRefEbRSt3mapIS5_jSt4lessIS5_ESaISt4pairIS6_jEEE()  ??:0
14 0x0000000001c88424 _ZN4llvm15RuntimeDyldImpl14loadObjectImplERKNS_6object10ObjectFileE()  ??:0
15 0x0000000001c9db64 _ZN4llvm14RuntimeDyldELF10loadObjectERKNS_6object10ObjectFileE()  ??:0
16 0x0000000001c814af _ZN4llvm11RuntimeDyld10loadObjectERKNS_6object10ObjectFileE()  ??:0
17 0x00000000001368e0 _ZN4llvm3orc30LegacyRTDyldObjectLinkingLayer20ConcreteLinkedObjectISt10shared_ptrINS_11RuntimeDyld13MemoryManagerEEE8finalizeEv()  /buildworker/worker/package_linux64/build/usr/include/llvm/ExecutionEngine/Orc/RTDyldObjectLinkingLayer.h:230
18 0x00000000001368e0 _ZNKSt10unique_ptrIN4llvm11RuntimeDyldESt14default_deleteIS1_EE3getEv()  /usr/local/include/c++/7.3.0/bits/unique_ptr.h:337
19 0x00000000001368e0 _ZNKSt10unique_ptrIN4llvm11RuntimeDyldESt14default_deleteIS1_EEptEv()  /usr/local/include/c++/7.3.0/bits/unique_ptr.h:331
20 0x00000000001368e0 _ZN4llvm3orc30LegacyRTDyldObjectLinkingLayer20ConcreteLinkedObjectISt10shared_ptrINS_11RuntimeDyld13MemoryManagerEEE8finalizeEv()  /buildworker/worker/package_linux64/build/usr/include/llvm/ExecutionEngine/Orc/RTDyldObjectLinkingLayer.h:235
21 0x000000000012e089 _ZN4llvm3orc30LegacyRTDyldObjectLinkingLayer15emitAndFinalizeEm()  /buildworker/worker/package_linux64/build/usr/include/llvm/ExecutionEngine/Orc/RTDyldObjectLinkingLayer.h:440
22 0x000000000012e089 _ZN4llvm5ErrorcvbEv()  /buildworker/worker/package_linux64/build/usr/include/llvm/Support/Error.h:237
23 0x000000000012e089 _ZN9JuliaOJIT9addModuleESt10unique_ptrIN4llvm6ModuleESt14default_deleteIS2_EE()  /buildworker/worker/package_linux64/build/src/jitlayers.cpp:500
24 0x000000000012f23b jl_add_to_ee()  /buildworker/worker/package_linux64/build/src/jitlayers.cpp:747
25 0x000000000012f23b jl_add_to_ee()  /buildworker/worker/package_linux64/build/src/jitlayers.cpp:747
26 0x000000000012f23b _Z20jl_finalize_functionN4llvm9StringRefE()  /buildworker/worker/package_linux64/build/src/jitlayers.cpp:755
27 0x0000000000041fb5 _ZL21getAddressForFunctionN4llvm9StringRefE()  /buildworker/worker/package_linux64/build/src/codegen.cpp:1414
28 0x000000000008882b jl_generate_fptr()  /buildworker/worker/package_linux64/build/src/codegen.cpp:1508
29 0x00000000000b3b3e jl_compile_method_internal()  /buildworker/worker/package_linux64/build/src/gf.c:1912
30 0x00000000000b3050 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2153
31 0x00000000000b3050 jl_apply_generic()  /buildworker/worker/package_linux64/build/src/gf.c:2322
32 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
33 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
34 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
35 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
36 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
37 0x00000000000c25cc jl_apply()  /buildworker/worker/package_linux64/build/src/julia.h:1700
38 0x00000000000c2e82 jl_f__apply_latest()  /buildworker/worker/package_linux64/build/src/builtins.c:693
39 0x0000000000db61db #invokelatest#1()  ./essentials.jl:712
40 0x000000000076f444 julia__start_2074()  ./client.jl:484
41 0x000000000076f586 jfptr__start_2075.clone_1()  julia:0
42 0x00000000000b340b _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2144
43 0x0000000000401932 jl_apply()  /buildworker/worker/package_linux64/build/ui/../src/julia.h:1700
44 0x0000000000401534 main()  /buildworker/worker/package_linux64/build/ui/repl.c:217
45 0x00000000000223d5 __libc_start_main()  ??:0
46 0x00000000004015d5 _start()  :0
===================

```

**Comment by kmdeck** [comment]
> I got the following error msg when making the docs locally.
> 
> ```
> [ Info: Populate: populating indices.
> ERROR: LoadError: `makedocs` encountered an error. Terminating build
> Stacktrace:
>  [1] [login2:166665:0:166665] Caught signal 7 (Bus error: nonexistent physical address)
> [login2:166665:0:166665] mmap(NULL, 4096, READ|WRITE, PRIVATE|ANON) failed: Cannot allocate memory
> [login2:166665:0] Caught signal 11 (Segmentation fault)
> ==== backtrace ====
>  2 0x000000000006ba2c mxm_handle_error()  /var/tmp/OFED_topdir/BUILD/mxm-3.7.3111/src/mxm/util/debug/debug.c:641
>  3 0x000000000006bf7c mxm_error_signal_handler()  /var/tmp/OFED_topdir/BUILD/mxm-3.7.3111/src/mxm/util/debug/debug.c:616
>  4 0x0000000000036280 killpg()  ??:0
>  5 0x000000000006b8a0 mxm_debug_get_line_info()  /var/tmp/OFED_topdir/BUILD/mxm-3.7.3111/src/mxm/util/debug/debug.c:340
>  6 0x00000000000472da backtrace_next()  ??:0
>  7 0x00000000000483bf ucs_handle_error()  ??:0
>  8 0x0000000000048750 ucs_handle_error()  ??:0
>  9 0x00000000000488a2 ucs_handle_error()  ??:0
> 10 0x000000000000f5d0 _L_unlock_13()  funlockfile.c:0
> 11 0x000000000015940b __memmove_ssse3_back()  :0
> 12 0x0000000001c830b5 _ZN4llvm15RuntimeDyldImpl11emitSectionERKNS_6object10ObjectFileERKNS1_10SectionRefEb()  ??:0
> 13 0x0000000001c838d5 _ZN4llvm15RuntimeDyldImpl17findOrEmitSectionERKNS_6object10ObjectFileERKNS1_10SectionRefEbRSt3mapIS5_jSt4lessIS5_ESaISt4pairIS6_jEEE()  ??:0
> 14 0x0000000001c88424 _ZN4llvm15RuntimeDyldImpl14loadObjectImplERKNS_6object10ObjectFileE()  ??:0
> 15 0x0000000001c9db64 _ZN4llvm14RuntimeDyldELF10loadObjectERKNS_6object10ObjectFileE()  ??:0
> 16 0x0000000001c814af _ZN4llvm11RuntimeDyld10loadObjectERKNS_6object10ObjectFileE()  ??:0
> 17 0x00000000001368e0 _ZN4llvm3orc30LegacyRTDyldObjectLinkingLayer20ConcreteLinkedObjectISt10shared_ptrINS_11RuntimeDyld13MemoryManagerEEE8finalizeEv()  /buildworker/worker/package_linux64/build/usr/include/llvm/ExecutionEngine/Orc/RTDyldObjectLinkingLayer.h:230
> 18 0x00000000001368e0 _ZNKSt10unique_ptrIN4llvm11RuntimeDyldESt14default_deleteIS1_EE3getEv()  /usr/local/include/c++/7.3.0/bits/unique_ptr.h:337
> 19 0x00000000001368e0 _ZNKSt10unique_ptrIN4llvm11RuntimeDyldESt14default_deleteIS1_EEptEv()  /usr/local/include/c++/7.3.0/bits/unique_ptr.h:331
> 20 0x00000000001368e0 _ZN4llvm3orc30LegacyRTDyldObjectLinkingLayer20ConcreteLinkedObjectISt10shared_ptrINS_11RuntimeDyld13MemoryManagerEEE8finalizeEv()  /buildworker/worker/package_linux64/build/usr/include/llvm/ExecutionEngine/Orc/RTDyldObjectLinkingLayer.h:235
> 21 0x000000000012e089 _ZN4llvm3orc30LegacyRTDyldObjectLinkingLayer15emitAndFinalizeEm()  /buildworker/worker/package_linux64/build/usr/include/llvm/ExecutionEngine/Orc/RTDyldObjectLinkingLayer.h:440
> 22 0x000000000012e089 _ZN4llvm5ErrorcvbEv()  /buildworker/worker/package_linux64/build/usr/include/llvm/Support/Error.h:237
> 23 0x000000000012e089 _ZN9JuliaOJIT9addModuleESt10unique_ptrIN4llvm6ModuleESt14default_deleteIS2_EE()  /buildworker/worker/package_linux64/build/src/jitlayers.cpp:500
> 24 0x000000000012f23b jl_add_to_ee()  /buildworker/worker/package_linux64/build/src/jitlayers.cpp:747
> 25 0x000000000012f23b jl_add_to_ee()  /buildworker/worker/package_linux64/build/src/jitlayers.cpp:747
> 26 0x000000000012f23b _Z20jl_finalize_functionN4llvm9StringRefE()  /buildworker/worker/package_linux64/build/src/jitlayers.cpp:755
> 27 0x0000000000041fb5 _ZL21getAddressForFunctionN4llvm9StringRefE()  /buildworker/worker/package_linux64/build/src/codegen.cpp:1414
> 28 0x000000000008882b jl_generate_fptr()  /buildworker/worker/package_linux64/build/src/codegen.cpp:1508
> 29 0x00000000000b3b3e jl_compile_method_internal()  /buildworker/worker/package_linux64/build/src/gf.c:1912
> 30 0x00000000000b3050 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2153
> 31 0x00000000000b3050 jl_apply_generic()  /buildworker/worker/package_linux64/build/src/gf.c:2322
> 32 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
> 33 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
> 34 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
> 35 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
> 36 0x00000000000b3063 _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2158
> 37 0x00000000000c25cc jl_apply()  /buildworker/worker/package_linux64/build/src/julia.h:1700
> 38 0x00000000000c2e82 jl_f__apply_latest()  /buildworker/worker/package_linux64/build/src/builtins.c:693
> 39 0x0000000000db61db #invokelatest#1()  ./essentials.jl:712
> 40 0x000000000076f444 julia__start_2074()  ./client.jl:484
> 41 0x000000000076f586 jfptr__start_2075.clone_1()  julia:0
> 42 0x00000000000b340b _jl_invoke()  /buildworker/worker/package_linux64/build/src/gf.c:2144
> 43 0x0000000000401932 jl_apply()  /buildworker/worker/package_linux64/build/ui/../src/julia.h:1700
> 44 0x0000000000401534 main()  /buildworker/worker/package_linux64/build/ui/repl.c:217
> 45 0x00000000000223d5 __libc_start_main()  ??:0
> 46 0x00000000004015d5 _start()  :0
> ===================
> ```



I see this error in the build in CI:
https://github.com/CliMA/ClimateMachine.jl/pull/1560/checks?check_run_id=1167809390#step:8:133

is it because init is referred to before it is defined?
https://github.com/CliMA/ClimateMachine.jl/blob/8566a5a34abd8ce053166ceaaacc4afb59d20fca/tutorials/Atmos/densitycurrent.jl#L55

(defined on line 120)

**Comment by kmdeck** [comment]
edit - not sure if that link to the error worked...
┌ Warning: 'init' is not unique in src/generated/Atmos/densitycurrent.md.

is what I was referring to.

**Comment by akshaysridhar** [comment]
> edit - not sure if that link to the error worked...
> ┌ Warning: 'init' is not unique in src/generated/Atmos/densitycurrent.md.
> 
> is what I was referring to.

There was a conflict in the `init` reference usage, modified it to strictly point to the init-condition function now

**Review by jiahe23 — APPROVED** [review]

**Pull request merged**